python/pyarrow/serialization.py

```diff
@@ -77,10 +78,10 @@ def _deserialize_default_dict(data):
     # python_to_arrow.cc)
 
     def _serialize_numpy_array(obj):
-        return obj.tolist(), obj.dtype.str
+        return pickle.dumps(obj)
```

Comment:
This could be made even more efficient by converting this to a `Buffer` so it's sent in a separate sidecar (for zero copy until `pickle.loads` is called). I will play around with this and see what the performance is like